index.ts

```diff
@@ -539,73 +601,85 @@ class ServerlessCustomDomain {
             await this.apigateway.updateBasePathMapping(params).promise();
             this.serverless.cli.log("Updated basepath mapping.");
         } catch (err) {
-            this.logIfDebug(err);
+            this.logIfDebug(err, domain.givenDomainName);
             throw new Error(`Error: Unable to update basepath mapping.\n`);
         }
     }
 
     /**
      * Gets rest API id from CloudFormation stack
      */
-    public async getRestApiId(): Promise<string> {
+    public async getApiId(domain: DomainConfig): Promise<string> {
         if (this.serverless.service.provider.apiGateway && this.serverless.service.provider.apiGateway.restApiId) {
             this.serverless.cli.log(`Mapping custom domain to existing API
                 ${this.serverless.service.provider.apiGateway.restApiId}.`);
             return this.serverless.service.provider.apiGateway.restApiId;
         }
+
         const stackName = this.serverless.service.provider.stackName ||
-            `${this.serverless.service.service}-${this.stage}`;
+            `${this.serverless.service.service}-${domain.stage}`;
+
+        let LogicalResourceId = "ApiGatewayRestApi";
+        if (domain.apiType === Globals.apiTypes.http) {
+            LogicalResourceId = "HttpApi";
+        } else if (domain.apiType === Globals.apiTypes.rest) {
```

Comment:
I missed the new `apiType` config when trying this out and noticed it tried to set up a `WebsocketsApi`. It's caused by comparing to the wrong type here.

```suggestion
        } else if (domain.apiType === Globals.apiTypes.websocket) {
```

Reply:
Gah, you are absolutely right. I originally had it hardcoded to `"WEBSOCKET"` but just messed up when I flipped it to use the Globals object and typed the wrong thing.

It should be fixed now.

Reply:
Saw a few other places and fixed them as well, it looks like it was errant find and replace.

Reply:
Also working on adding some unit tests to catch this.

Reply:
Unit tests added that would/should have caught this.